src/operand/__test__/unit/reference.test.ts

```diff
@@ -66,6 +67,11 @@ describe('Operand - Value', () => {
       ['RefA{RefA}', undefined],
       ['RefB.{RefA}', undefined],
       ['Ref{RefB}', undefined],
+      // Data type parsing
+      ['RefH[{RefA}].sub{RefD}.(Number)', 2],
+      ['RefA.(String)', '1'],
+      ['RefJ.(String)', '1'],
+      ['RefJ.(Number)', 1],
```

Comment:
For familiarity, go the way of Typescript instead? e.g. `<Number>RefJ`.  It's also one less character (not that it really matters).

Reply:
Idea is for illogical's DSL to be language agnostic considering that there will probably be implementations in other languages as well.

Reply:
Sure, but not sure if I've seen casting done as `.(type)` in languages.

Reply:
http://rigaux.org/language-study/syntax-across-languages.html#TpsCast

`(Type) $ref`?

Reply:
:) https://golang.org/ref/spec#Type_assertions

Reply:
> Idea is for illogical's DSL to be language agnostic considering that there will probably be implementations in other languages as well.

💯 ... though really the format does not matter, if easy to understand / implement

Reply:
I stand corrected!  I'll leave it to you folks to duel it out.

Reply:
@theslyone you can create a quick pool in the #carriers-channel to get votes :D

Reply:
> @theslyone you can create a quick pool in the #carriers-channel to get votes :D

Hopefully will wouldn't have to drag this decision to slack, i personally think `$ref.(Type)`, `$ref.<Type>`, `<Type>.$ref`, `(Type)$ref`, `<Number>$ref` and all it's variants work fine and we should just stick to one and have the readme/docs updated appropriately.

Reply:
Yeah. I have to strong preference for any of those, so I will it keep it up to your decision.